lib/gstvkvideoparser.cpp

```diff
@@ -89,49 +92,44 @@ process_messages (GstVkVideoParser * self)
   }
 }
 
-static void
-gst_vk_video_parser_dispose (GObject * object)
+void GstVkVideoParser::Dispose ()
 {
-  GstVkVideoParser *self = GST_VK_VIDEO_PARSER (object);
+  
   GstMessage *msg;
 
-  gst_harness_teardown (self->parser);
+  gst_harness_teardown (m_parser);
 
   /* drain bus after bin unref */
-  while ((msg = gst_bus_pop (self->bus))) {
-    GST_DEBUG_OBJECT (self, "%s", GST_MESSAGE_TYPE_NAME (msg));
+  while ((msg = gst_bus_pop (this->m_bus))) {
+    GST_DEBUG_OBJECT (this, "%s", GST_MESSAGE_TYPE_NAME (msg));
     gst_message_unref (msg);
   }
 
-  gst_object_unref (self->bus);
-
-  G_OBJECT_CLASS (gst_vk_video_parser_parent_class)->dispose (object);
+  gst_object_unref (this->m_bus);
 }
 
-static void
-gst_vk_video_parser_constructed (GObject * object)
+bool GstVkVideoParser::Build ()
```

Comment:
What about using the C++ constructor for this?

Reply:
To check in GstVkVideoDecoderParser::Initialize that the initialization succeeded